cmd/restic/cmd_rewrite.go

```diff
@@ -203,7 +240,7 @@ func filterAndReplaceSnapshot(ctx context.Context, repo restic.Repository, sn *r
 		return true, nil
 	}
 
-	if filteredTree == *sn.Tree && newMetadata == nil {
+	if filteredTree == *sn.Tree && newMetadata == nil && sn.Summary == nil {
```

Comment:
This breaks the "snapshot %v not modified" check if `rewrite --exclude ...` or `repair snapshots` are used on snapshots that already have a summary section.